anchor/common/ssv_types/src/cluster.rs

```diff
@@ -0,0 +1,53 @@
+use crate::OperatorId;
+use crate::Share;
+use derive_more::{Deref, From};
+use types::{Address, Graffiti, PublicKey};
+
+/// Unique identifier for a cluster
+#[derive(Clone, Copy, Debug, Default, Eq, PartialEq, Hash, From, Deref)]
+pub struct ClusterId(pub u64);
+
+/// A Cluster is a group of Operators that are acting on behalf of a Validator
+#[derive(Debug, Clone)]
+pub struct Cluster {
+    /// Unique identifier for a Cluster
+    pub cluster_id: ClusterId,
+    /// All of the members of this Cluster
+    pub cluster_members: Vec<ClusterMember>,
+    /// The number of faulty operator in the Cluster
+    pub faulty: u64,
+    /// If the Cluster is liquidated or active
+    pub liquidated: bool,
+    /// Metadata about the validator this committee represents
+    pub validator_metadata: ValidatorMetadata,
+}
+
+/// A member of a Cluster. This is just an Operator that holds onto a share of the Validator key
+#[derive(Debug, Clone)]
+pub struct ClusterMember {
+    /// Unique identifier for the Operator this member represents
+    pub operator_id: OperatorId,
+    /// Unique identifier for the Cluster this member is a part of
+    pub cluster_id: ClusterId,
+    /// The Share this member is responsible for
+    pub share: Share,
+}
+
+/// Index of the validator in the validator registry.
+#[derive(Clone, Copy, Debug, Default, Eq, PartialEq, Hash, From, Deref)]
+pub struct ValidatorIndex(pub usize);
+
+/// General Metadata about a Validator
+#[derive(Debug, Clone)]
+pub struct ValidatorMetadata {
```

Comment:
is there a reason we've pulled out the committee ID and domaintype from here?

Reply:
I think this type is supposed to basically represent the metadata of the validator operated by the cluster, not the cluster itself, so I think I like leaving it out here.

Reply:
Yep, this is just about the Validator the cluster is acting on behalf of. Can be stored in database as ClusterID => ValidatorMetadata so this info still gets included. As for the domain, I left it off because I believe it is being included via lighthouse types, but I have to double check to confirm what it gets included through.